crates/indexer/src/insert_l2_block.rs

```diff
@@ -0,0 +1,284 @@
+                .into_par_iter()
+                .map(|l| DbLog::try_from_log(l, i as i64))
+                .collect();
+            (DbTransaction::try_from(web3_tx_with_logs.tx), db_logs)
+        })
+        .collect::<(Vec<_>, Vec<_>)>();
+    let txs = txs.into_iter().collect::<Result<Vec<_>>>()?;
+    let logs = logs.into_iter().collect::<Result<Vec<_>>>()?;
+    let logs = logs.into_iter().flatten().collect::<Vec<_>>();
+
+    let logs_len = logs.len();
+    let txs_len = txs.len();
+
+    let logs_slice = logs
+        .into_iter()
+        .chunks(INSERT_LOGS_BATCH_SIZE)
+        .into_iter()
+        .map(|chunk| chunk.collect())
+        .collect::<Vec<Vec<_>>>();
+
+    let mut txs_query_builder: QueryBuilder<Postgres> = QueryBuilder::new(
+                "INSERT INTO transactions
+                (hash, eth_tx_hash, block_number, block_hash, transaction_index, from_address, to_address, value, nonce, gas_limit, gas_price, input, v, r, s, cumulative_gas_used, gas_used, contract_address, exit_code) "
+            );
+
+    txs_query_builder
+        .push_values(txs, |mut b, tx| {
+            b.push_bind(tx.hash)
+                .push_bind(tx.eth_tx_hash)
+                .push_bind(tx.block_number)
+                .push_bind(tx.block_hash)
+                .push_bind(tx.transaction_index)
+                .push_bind(tx.from_address)
+                .push_bind(tx.to_address)
+                .push_bind(tx.value)
+                .push_bind(tx.nonce)
+                .push_bind(tx.gas_limit)
+                .push_bind(tx.gas_price)
+                .push_bind(tx.input)
+                .push_bind(tx.v)
+                .push_bind(tx.r)
+                .push_bind(tx.s)
+                .push_bind(tx.cumulative_gas_used)
+                .push_bind(tx.gas_used)
+                .push_bind(tx.contract_address)
+                .push_bind(tx.exit_code);
+        })
+        .push(" RETURNING id");
+
+    let mut tx_ids: Vec<i64> = vec![];
+
+    let query = txs_query_builder.build();
+    let rows: Vec<PgRow> = query.fetch_all(&mut (*pg_tx)).await?;
+    let mut ids = rows
+        .iter()
+        .map(|r| r.get::<i64, _>("id"))
+        .collect::<Vec<i64>>();
+    tx_ids.append(&mut ids);
+
+    let logs_querys = logs_slice
+            .into_par_iter()
+            .map(|db_logs| {
+                let mut logs_query_builder: QueryBuilder<Postgres> = QueryBuilder::new(
+                    "INSERT INTO logs
+                    (transaction_id, transaction_hash, transaction_index, block_number, block_hash, address, data, log_index, topics)"
+                );
+
+                // Get transaction id from preview insert returning
+                logs_query_builder.push_values(db_logs, |mut b, log| {
+                    // transaction_id in log is transaction_id_index now
+                    let transaction_id_index = log.transaction_id as usize;
+                    let transaction_id = tx_ids[transaction_id_index];
```

Comment:
Can we use `let transaction_id = tx_ids[log.transaction_index];` here?

Reply:
ok